### PR TITLE
add pad_to_buckets in evaluation for hpu performance

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

lm_eval evaluation enhancement

## Description

buckets = [64, 128, 256, 512, 1024, 2048, 4096, 8192]
This change will pad the input length to the nearest upper bound in buckets, which will avoid creating many graphs for different lengths to leverage the HPU accelerator.

- [x] `pad_to_buckets` is a new argument for `LMEvalParser`
- [x] auto device is enabled if `pad_to_buckets` is not set. 
- [x] add hpu evaluation support in example.
- [x] install evaluation when install neural_compressor_pt.

## Expected Behavior & Potential Risk

lm_eval example test gives the same result as before.

### CODE REVIEW

**Comment by xin3he** [comment]
https://inteltf-jenk.sh.intel.com/job/INC_LLM_accuracy/137/artifact/report.html

File: neural_compressor/evaluation/lm_eval/models/huggingface.py [hunk 1]
```diff
@@ -511,7 +514,6 @@ def _create_model(
                 if model_kwargs.get("load_in_4bit", None):
                     if model_kwargs.get("bnb_4bit_compute_dtype", None):
                         model_kwargs["bnb_4bit_compute_dtype"] = get_dtype(model_kwargs["bnb_4bit_compute_dtype"])
-            from neural_compressor.transformers import AutoModelForCausalLM
```
Comment (changwangss) [inline neural_compressor/evaluation/lm_eval/models/huggingface.py:514]: why remove this line? if you remove this line, woq model generated by inc will load failed.

Comment (xin3he) [reply neural_compressor/evaluation/lm_eval/models/huggingface.py:514]: Hi @changwangss  I think it's already import at the beginning
https://github.com/intel/neural-compressor/blob/b6b7d7c3c415d67976e054ab5ad5be6b5d5b460d/neural_compressor/evaluation/lm_eval/models/huggingface.py#L42

Comment (changwangss) [reply neural_compressor/evaluation/lm_eval/models/huggingface.py:514]: ok, approve.

**Review by changwangss — APPROVED** [review]

**Review by Kaihui-intel — APPROVED** [review]

**Pull request merged**